zerver/views/users.py

```diff
@@ -162,9 +172,30 @@ def _deactivate_user_profile_backend(
     user_profile: UserProfile,
     target: UserProfile,
     *,
+    is_spammer: Json[bool] = False,
+    message_delete_action: Json[MessageDeleteAction] | None = None,
     deactivation_notification_comment: str | None,
 ) -> HttpResponse:
-    do_deactivate_user(target, acting_user=user_profile)
+    message_delete_action_list: list[int] | None = None
+    if message_delete_action is not None:
+        message_delete_action_list = []
+        if message_delete_action.delete_public_stream_messages:
+            message_delete_action_list.append(Message.DELETE_PUBLIC_STREAM_MESSAGES)
+        if message_delete_action.delete_private_stream_messages:
+            message_delete_action_list.append(Message.DELETE_PRIVATE_STREAM_MESSAGES)
+        if message_delete_action.delete_direct_messages:
+            message_delete_action_list.append(Message.DELETE_DIRECT_MESSAGES)
+
+        # If no actions selected, default to no delete
+        if not message_delete_action_list:
+            message_delete_action_list = [Message.NO_DELETE_ACTION]
```

Comment:
I think it would be simpler to just pass the object as it is since all three cases are being handled separately. For case of no delete action, `delete_deactivated_user_messages` can just return early if all the fields are false.